hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/V2Migration.java

```diff
@@ -95,4 +99,13 @@ public static void v1GetObjectMetadataCalled() {
         + "will be changed as part of upgrading S3A to AWS SDK V2");
   }
 
+  /**
+   * Warns on use of getObjectMetadata.
+   */
+  public static void regionNotConfigured() {
+    WARN_ON_REGION_NOT_CONFIGURED.warn("A region has not been configured. Cross region support "
+            + "will be removed as part of upgrading S3A to AWS SDK V2. To avoid errors, set the "
+        + "bucket's region in {}.", AWS_REGION);
+  }
```

Comment:
How about passing region name to the method and let the method decide whether to warn?

```
  public static void warnIfRegionNotConfigured(String regionName) {
    if (StringUtils.isBlank(regionName)) {
      WARN_ON_REGION_NOT_CONFIGURED.warn("A region has not been configured. Cross region support "
          + "will be removed as part of upgrading S3A to AWS SDK V2. To avoid errors, set the "
          + "bucket's region in {}.", AWS_REGION);
    }
  }

```